agent/engine/docker_container_engine.go

```diff
@@ -726,7 +726,7 @@ func (dg *dockerGoClient) Stats(id string, ctx context.Context) (<-chan *docker.
 	go func() {
 		statsErr := client.Stats(options)
 		if statsErr != nil {
-			seelog.Warnf("Error retrieving stats for container %s: %v", id, statsErr)
+			seelog.Debugf("Error retrieving stats for container %s: %v", id, statsErr)
```

Comment:
I decided against doing that and leaving it up to the caller to handle the error. Because, eventually, the stream does get closed irrespective of the error printed here. Go-dockerclient ensures that. I think it makes more sense for the error to be surfaced via an error channel so that the caller can take the requisite action upon receiving the error in the channel (this is not part of this PR as it requires more thought and time). Does that make sense?